### PR TITLE
Convert Inherit Someone Dies Without Will flow tests

### PR DESCRIPTION
**Pull request by leenagupte**
Trello: https://trello.com/c/imrruScc

# What's changed and why?

Converts the Inherit if Someone Dies Without Will flow tests to use the node based testing approach introduced in [ADR-3][1].

Also removes the existing flow integration tests as they are no longer needed.

[1]: https://github.com/alphagov/smart-answers/blob/main/docs/arch/003-testing-flows.md

⚠️ This repo is Continuously Deployed: make sure you [follow the guidance](https://docs.publishing.service.gov.uk/manual/development-pipeline.html#merge-your-own-pull-request) ⚠️

### CODE REVIEW

File: test/support/flow_test_helper.rb [hunk 1]
```diff
@@ -154,12 +154,16 @@ def outcome_title
...
+      outcome_presenter.next_steps
+    end
+
     def outcome_body_text
       Nokogiri::HTML::DocumentFragment.parse(outcome_body).text
     end
 
     def outcome_text
-      [outcome_title, outcome_body_text].reject(&:blank?).join("\n")
+      [outcome_title, outcome_body_text, outcome_next_steps].reject(&:blank?).join("\n")
```
Comment (kevindew) [inline test/support/flow_test_helper.rb:166]: As `outcome_next_steps` is HTML could we have this as a `outcome_next_steps_text` method that strips the HTML for consistency?

(Just a quick drive-by comment as I know Rosa is doing a more thorough review)

Comment (leenagupte) [reply test/support/flow_test_helper.rb:166]: Oops! Missed this. I think was using `title` as a template. Will update and fix up.

File: test/flows/inherits_someone_dies_without_will_flow_test.rb [hunk 2]
```diff
@@ -0,0 +1,558 @@
...
+      should "have a next node of parents? for a 'no' response if has partner and region is northern-ireland" do
+        add_responses region?: "northern-ireland",
+                      estate_over_250000?: "yes"
+        assert_next_node :parents?, for_response: "no"
+      end
+
+      should "have a next node of outcome_66 for a 'yes' response if has no partner and region is northern-ireland" do
+        add_responses region?: "northern-ireland",
+                      partner?: "no"
+        assert_next_node :outcome_66, for_response: "yes"
```
Comment (GDSNewt) [inline test/flows/inherits_someone_dies_without_will_flow_test.rb:175]: I'd have loved it if this outcome was just a palpatine gif

**Review by GDSNewt — APPROVED** [review]
Hi Leena, apologies for the delay in getting this review out I wanted to be extra thorough, so I flowcharted the flow out in a graph. I couldn't find any impossible to reach nodes or anything like that and the flow does correspond with the tests. 

I like how you tested the partials for content, it seems very obvious having seen how you did it but I did wonder initially how all these outcomes were going to be tested. 

Looks good to merge to me :shipit:
